packages/wallets/bravewallet/src/adapter.ts

```diff
@@ -0,0 +1,268 @@
+import {
+    BaseMessageSignerWalletAdapter,
+    EventEmitter,
+    scopePollingDetectionStrategy,
+    SendTransactionOptions,
+    WalletAccountError,
+    WalletConnectionError,
+    WalletDisconnectedError,
+    WalletDisconnectionError,
+    WalletError,
+    WalletName,
+    WalletNotConnectedError,
+    WalletNotReadyError,
+    WalletPublicKeyError,
+    WalletReadyState,
+    WalletSignTransactionError,
+    WalletWindowClosedError,
+} from '@solana/wallet-adapter-base';
+import { Connection, PublicKey, SendOptions, Transaction, TransactionSignature } from '@solana/web3.js';
+
+interface BraveWalletEvents {
+    connect(...args: unknown[]): unknown;
+    disconnect(...args: unknown[]): unknown;
+}
+
+interface BraveWallet extends EventEmitter<BraveWalletEvents> {
+    isBraveWallet?: boolean;
+    publicKey?: { toBytes(): Uint8Array };
+    isConnected: boolean;
+    signTransaction(transaction: Transaction): Promise<Transaction>;
+    signAllTransactions(transactions: Transaction[]): Promise<Transaction[]>;
+    signAndSendTransaction(
+        transaction: Transaction,
+        options?: SendOptions
+    ): Promise<{ signature: TransactionSignature }>;
+    signMessage(message: Uint8Array): Promise<{ signature: Uint8Array }>;
+    connect(): Promise<void>;
+    disconnect(): Promise<void>;
+    _handleDisconnect(...args: unknown[]): unknown;
+}
+
+interface BraveWindow extends Window {
+    solana?: BraveWallet;
```

Comment:
Using `window.solana` isn't permitted for new adapters, and we are working with the few wallets that do use it to migrate all adapters away from it because conflicts over the variable make detection of available wallets impossible. Please consider changing your extension to `window.brave` or similar so it doesn't conflict with others!

Reply:
@jordansexton we're purposely exposing window.solana so that there is Phantom wallet compatibility for sites that don't use this adapter or don't update it, but we'll recommend that people use this adapter that we're trying to land here.

Note that we have a setting for controlling if we expose our provider or not. And by default if the user has Phantom it will use Phantom.

Reply:
I understand, and I can't tell you to not write to `window.solana`, but we're not accepting new adapters that use `window.solana`, and we're actively migrating away from this.

We've told this to all other wallets for almost a year, and there are a few good reasons for this restriction, so we must remain consistent about this.

Once the wallet standard provides a location for wallets to attach to, any supporting app will detect Brave Wallet without needing to make a change to support it specifically.

Reply:
fixed in https://github.com/solana-labs/wallet-adapter/pull/445/commits/db4c9ed630324c2db52183bebd76b75692bbd016